index.bs

```diff
@@ -362,59 +361,102 @@ The client facilitates these security measures by providing correct origins and
 operation. Since this is an integral part of the WebAuthn security model, user agents MUST only expose this API to callers in
 [=secure contexts=].
 
-The Web Authentication API is defined by the union of the Web IDL fragments presented in the following sections. A combined IDL listing is given in the [[#idl-index]]. The API is defined as a part of the {{Navigator}} interface:
-
-
-<pre class="idl">
-    partial interface Navigator {
-        readonly attribute WebAuthentication authentication;
-    };
-</pre>
+The Web Authentication API is defined by the union of the Web IDL fragments presented in the following sections. A combined IDL
+listing is given in the [[#idl-index]].
 
+## <dfn interface>ScopedCredential</dfn> Interface ## {#iface-scopedcredential}
 
-## <dfn interface>WebAuthentication</dfn> Interface ## {#iface-credential}
+The {{ScopedCredential}} interface inherits from {{Credential}} [[!CREDENTIAL-MANAGEMENT-1]], and contains the attributes that
+are returned to the caller when a new credential is created, or a new assertion is requested.
 
 <xmp class="idl">
     [SecureContext]
-    interface WebAuthentication {
-        Promise<ScopedCredentialInfo> makeCredential(
+    interface ScopedCredential : Credential {
+        readonly attribute ArrayBuffer           rawID;
+        readonly attribute AuthenticatorResponse response;
+
+        static Promise<ScopedCredential> create(
             Account                              accountInformation,
             sequence<ScopedCredentialParameters> cryptoParameters,
             BufferSource                         attestationChallenge,
             optional ScopedCredentialOptions     options
         );
-
-        Promise<AuthenticationAssertion> getAssertion(
-            BufferSource                    assertionChallenge,
-            optional AssertionOptions       options
-        );
     };
 </xmp>
+<dl dfn-type="attribute" dfn-for="ScopedCredential">
+    :   {{Credential/id}}
+    ::  {{ScopedCredential}} overrides {{Credential}}'s getter for this attribute, returning the [=base64url encoding=] of the
+        data contained in the object's {{ScopedCredential/[[identifier]]}} slot.
+
+    :   {{ScopedCredential/rawID}}
+    ::  This attribute returns the {{ArrayBuffer}} contained in the {{ScopedCredential/[[identifier]]}} internal slot.
```

Comment:
Typed arrays are "Integer Indexed Exotic Objects".  Per https://tc39.github.io/ecma262/#sec-integer-indexed-exotic-objects that means their integer-named properties are writable and non-configurable, so you can't make them non-writable.

But that's a red herring anyway, because you can just create a _new_ typed array of your own using the same backing arraybuffer.

If you actually perform a set on a typed array, you end up in https://tc39.github.io/ecma262/#sec-integer-indexed-exotic-objects-set-p-v-receiver and thence in https://tc39.github.io/ecma262/#sec-integerindexedelementset which will do some validation and call https://tc39.github.io/ecma262/#sec-setvalueinbuffer

There are no provisions for readonlyness of arraybuffers and therefore no provisions for readonlyness of typed arrays.  The bluetooth spec is basically unimplementable as written.

Oh, and adding such provisions is likely to cause prohibitive performance regressions for code that uses typed arrays.  Which means JS engine implementors will push back very hard on attempts to add something like this.  They have before.

The right way to expose binary data that you don't want mutated is to expose a copy.